### PR TITLE
Support for custom formatting and ordering of subtitles

### PR DESCRIPTION
**Pull request by nevir**
This allows for users to customize the format of the subtitle by deciding which information is included, and the order of those items.

We also add support for displaying the username ("additional_info") that is associated with each login

### CODE REVIEW

File: Workflow/info.plist [hunk 1]
```diff
@@ -2043,9 +2044,9 @@ Uncommon but useful actions, such as toggling vaults, can be accessed via `:1pex
 		<string>1</string>
 		<key>logins_only</key>
 		<string>0</string>
+		<key>subtitle_items</key>
+		<string>username, url, vault, opAccount</string>
 	</dict>
-	<key>variablesdontexport</key>
-	<array/>
```
Comment (nevir) [inline Workflow/info.plist:2048]: Alfred drops this when I edit via the UI

**Comment by vitorgalvao** [comment]
Thank you for the PR. I’ve been thinking about it since last night, and while I like the idea it feels like too much customisation. The target user for this Workflow is very broad and having a Workflow Environment Variable which needs to be *specific* words separated by commas is tricky and doesn’t seem worth the extra complexity in this case.

Making `additional_information` available is a particularly delicate matter. The field wasn’t available in the previous third-party integration, and for good reason: usernames can themselves be sensitive information and in the case of Secure Notes, that field shows the first line. We can make the argument that because it’s an option, the Workflow can give the user the flexibility of which data to keep, but saving sensitive data should not be turned on by default. Given the broad range of users (many of which not computer experts) this Workflow is geared for and the potential sensitivity of the data, I’m reticent to even provide the option.

It seems the impetus for this feature was hiding the URLs. Can you help me understand your motivation for that? Why do you want to hide them? Is it purely an aesthetic choice?

**Comment by nevir** [comment]
Yeah, that makes a lot of sense - won't hurt my feelings if you don't want
to merge this :)

I have two motivations:

* Hiding URLs because some login URLs are very long (and it is the least
useful piece of information for me). Truncating to just domain would
probably help a lot here

* Username is critical information for me - at work I have a number of
cases where I have to juggle multiple accounts for the same site, and
prefer to disambiguate via that rather than entry title (1Password's UI
does a very good job on Mac and Android at least by putting usernames front
and center for these logins)

**Comment by vitorgalvao** [comment]
> Truncating to just domain would probably help a lot here

I’ve added a new `hostnames_only` option to do just that. It’s on by default. I still have to document it, [but it’s working](https://github.com/alfredapp/1password-workflow/files/8642998/1Password.alfredworkflow.zip). I’ll release it soon, but that version will update to it.

As for the username, I’ll give it further thought but at least for now it won’t be in.

**Pull request closed without merging**